examples/Cloud_Robotics/singletask_learning_bench/Semantic_Segmentation/testalgorithms/basemofel.py

```diff
@@ -0,0 +1,175 @@
+from RFNet.utils.lr_scheduler import LR_Scheduler
+from RFNet.train import Trainer
+from RFNet.eval import Validator
+from RFNet.utils.saver import Saver
+from RFNet.eval import load_my_state_dict
+import RFNet.train_config as train_cfgs
+import RFNet.eval_config as valid_cfgs
+
+logging.disable(logging.WARNING)
+
+__all__ = ["BaseModel"]
+
+os.environ['BACKEND_TYPE'] = 'TORCH'
+
+
+@ClassFactory.register(ClassType.GENERAL, alias="RFNet")
+class BaseModel:
+
+    def __init__(self, **kwargs):
+        self.train_cfgs = train_cfgs
+        self.valid_cfgs = valid_cfgs
+
+        self.train_cfgs.lr = kwargs.get("learning_rate", 0.0001)
+        self.train_cfgs.momentum = kwargs.get("momentum", 0.9)
+        self.train_cfgs.epochs = kwargs.get("epochs", 2)
+        self.train_cfgs.checkname = ''
+        self.train_cfgs.batch_size = 4
+        self.train_cfgs.depth = True
+
+        os.environ["MODEL_NAME"] = "model_best_mapi_only.pth"
+
+        self.train_cfgs.gpu_ids = ['cuda:0']
+        self.valid_cfgs.gpu_ids = ['cuda:0']
+        self.valid_cfgs.depth = True
+
+        self.validator = Validator(self.valid_cfgs)
+        self.trainer = Trainer(self.train_cfgs)
+        self.trainer.saver = Saver(self.train_cfgs)
+        self.checkpoint_path = self.load(
+            Context.get_parameters("base_model_url"))
+
+    def train(self, train_data, valid_data=None, **kwargs):
+        if train_data is None or train_data.x is None or train_data.y is None:
+            raise Exception("Train data is None.")
+
+        self.trainer.train_loader, self.trainer.val_loader, _, _ = \
+            make_data_loader(
+                self.trainer.args, train_data=train_data, valid_data=valid_data, **kwargs)
+
+        # Define lr scheduler
+        self.trainer.scheduler = LR_Scheduler(self.trainer.args.lr_scheduler, self.trainer.args.lr,
+                                              self.trainer.args.epochs, len(self.trainer.train_loader))
+        print("Total epoches:", self.trainer.args.epochs)
+
+        for epoch in range(self.trainer.args.start_epoch, self.trainer.args.epochs):
+            if epoch == 0 and self.trainer.val_loader:
+                self.trainer.validation(epoch)
+            self.trainer.training(epoch)
+
+            if self.trainer.args.no_val and \
+                    (epoch % self.trainer.args.eval_interval == (self.trainer.args.eval_interval - 1)
+                     or epoch == self.trainer.args.epochs - 1):
+                # save checkpoint when it meets eval_interval or the training finished
+                is_best = False
+                self.checkpoint_path = self.trainer.saver.save_checkpoint({
+                    'epoch': epoch + 1,
+                    'state_dict': self.trainer.model.state_dict(),
+                    'optimizer': self.trainer.optimizer.state_dict(),
+                    'best_pred': self.trainer.best_pred,
+                }, is_best)
+
+            # if not self.trainer.args.no_val and \
```

Comment:
Considering Clean Code: remove redundant codes instead of making them as comment